openbb_platform/platform/core/openbb_core/app/model/obbject.py

```diff
@@ -183,37 +183,26 @@ def to_numpy(self) -> ndarray:
         """Convert results field to numpy array."""
         return self.to_dataframe().reset_index().to_numpy()
 
-    def to_dict(self) -> Dict[str, List]:
-        """Convert results field to list of values.
+    def to_dict(self) -> Union[Dict[str, Any], List[Dict[str, Any]]]:
+        """Convert results field Data models to dict.
 
         Returns
         -------
-        Dict[str, List]
-            Dictionary of lists.
+        Union[Dict[str, Any], List[Dict[str, Any]]]
+            Dict or list of dicts.
         """
-        if isinstance(self.results, dict) and all(
-            isinstance(v, dict) for v in self.results.values()
-        ):
-            results: Dict[str, List] = {}
-            for _, v in self.results.items():
-                for kk, vv in v.items():
-                    if kk not in results:
-                        results[kk] = []
-                    results[kk].append(vv)
-
-            return results
-
-        df = self.to_dataframe().reset_index()  # type: ignore
-        results = {}
-        for field in df.columns:
-            f = df[field].tolist()
-            results[field] = f[0] if len(f) == 1 else f
-
-        # remove index from results
-        if "index" in results:
-            del results["index"]
-
-        return results
+
+        def nested_model_to_dict(item: Any) -> Any:
+            if hasattr(item, "model_dump"):
+                return item.model_dump()
+            elif isinstance(item, list):
+                return [nested_model_to_dict(i) for i in item]
+            elif isinstance(item, dict):
+                return {k: nested_model_to_dict(v) for k, v in item.items()}
+
+            return item
+
+        return nested_model_to_dict(self.results)
```

Comment:
Does it cover your use case if we just implement in one line?

```
def to_dict(self) -> Dict
    ...
    return self.model_dump()["results"]
```

Reply:
🤦🏻‍♂️Lmao, this what happens when doing a @DidierRLopes task at 1am.

Yeah this for sure the best way to go!!